pkg/controller/swiftproxy/swiftproxy_controller.go

```diff
@@ -134,7 +134,7 @@ func (r *ReconcileSwiftProxy) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	servicePortsMap := map[int32]string{int32(swiftProxy.Spec.ServiceConfiguration.ListenPort): ""}
-	svc := r.kubernetes.Service(request.Name+"-swiftproxy", core.ServiceTypeLoadBalancer, servicePortsMap, contrail.SwiftProxyInstanceType, swiftProxy)
+	svc := r.kubernetes.Service(request.Name+"-swiftproxy", swiftProxy.Spec.ServiceConfiguration.SwiftServiceType, servicePortsMap, contrail.SwiftProxyInstanceType, swiftProxy)
```

Comment:
I think that we need to start with the default value set to core.ServiceTypeLoadBalancer. When SwiftServiceType is empty string we should use ServiceTypeLoadBalancer. If we pass empty string here we will get ClusterIP service. That might have an effect on other deployments.